src/BenchmarkDotNet/Portability/RuntimeInformation.cs

```diff
@@ -160,18 +160,14 @@ internal static CpuInfo GetCpuInfo()
             return null;
         }
 
-        internal static string GetNetCoreVersion() => InDocker ? GetDockerNetCoreVersion() : GetBaseNetCoreVersion();
+        internal static Version GetNetCoreVersion()
+        {
+            var frameworkName = Assembly.GetEntryAssembly()?.GetCustomAttribute<TargetFrameworkAttribute>()?.FrameworkName;
```

Comment:
> Please note that in some cases Assembly.GetExecutingAssembly() will yield a different result compared to the previous logic (potentially breaking user benchmarks).
This happens for instance when running unit tests. The target framework attribute of the executing assembly (testhost.dll) has the value .NETCoreApp,Version=v1.0, hence the new logic will detect version 1.0.

@jasper-d I tested that and you are right. However, I can see that `GetExecutingAssembly` returns valid result. I tested it for xunit runner and standalone netcoreapp. So please change `GetEntryAssembly` to `GetExecutingAssembly`

big thanks for testing this scenario and mentioning this!

```suggestion
            var frameworkName = Assembly.GetExecutingAssembly()?.GetCustomAttribute<TargetFrameworkAttribute>()?.FrameworkName;
```

Reply:
@adamsitnik Sorry, I meant GetEntryAssembly when I wrote GetExecutingAssembly. 😕

Anyway, according to my understanding of GetExecutingAssembly and a quick test, it's even worse. It will return the actual assembly it is called from, so in this case it will always return the BDN assembly.

The following test will fail when running inside a netcoreapp2.2 test project. When using GetEntryAssembly `result` will be 1.0 (testhost.dll), when using GetExecutingAssembly `result` will be 2.1 (BenchmarkDotNet.dll).

```
        [Fact]
        public void GetTargetFrameworkVersionReturns22()
        {
            var currentEnvironment = new FrameworkName(Assembly.GetExecutingAssembly().GetCustomAttribute<TargetFrameworkAttribute>().FrameworkName).Version; //2.2

            var result = BenchmarkDotNet.Portability.RuntimeInformation.GetTargetFrameworkVersion();

            Assert.True(new Version(2, 2) == currentEnvironment, "Assert environment"); //Sanity check, passes always
            Assert.Equal(currentEnvironment, result); //Fails
        }
```

Reply:
I would propose to fix the original implementation

Reply:
@jasper-d you are right. Now we know how both solutions work, but none of them  covers 100% of cases ;(

Reply:
@adamsitnik So how do we move forward from here? I would propose to close this PR and make a new one that fixes the previous logic, taking into account the varying directory names. Maybe logging a warning to inform users about failed version detection would be helpful as well, just in case.

I still like the GetEntryAssembly approach because it makes the code significantly simpler and removes the need for special handling of containerised applications (would probably worth to test rkt as well). 
I don't know how other people use BDN, but I assume that for the vast majority the entry assembly will be a console application they wrote for their benchmarking and not some other DLL (like testhost). Hence, I think it would be a viable option to update this PR with your change requests (testing will be difficult though). Then come back to it when a new major version of BDN with other breaking changes is going to be released. In cases where the entry assembly is different, users could then use explicit configurations.

Your thoughts?

Reply:
@adamsitnik Any opinion?